gen_test.py

```diff
@@ -0,0 +1,301 @@
+        result.update(dict(zip(names, input)))
+        result.update({"generator": "gen_test_common_substring_utf8"})
+        results.append(result)
+
+        texts = gen_test_common_subsequence_utf8(*input)
+        result = benchmark_diff(*texts, timelimit)
+        result.update(dict(zip(names, input)))
+        result.update({"generator": "gen_test_common_subequence_utf8"})
+        results.append(result)
+
+    batch_end = time.time()
+    print(f"Batch time: {batch_end - batch_start:.2f}")
+    return results
+
+def quick_test():
+    texts = gen_test_common_subsequence_utf8(12345, 15022, 1)
+    result = benchmark_diff(*texts, timelimit=True)
+    speedup = f"{float(result['delta_old']) / float(result['delta_new']):.2f}"
+    print(f"text1_len: 12345; text2_len: 15022; time_new: {result['delta_new']}; time_old: {result['delta_old']}; speedup: {speedup}")
+
+    texts = gen_test_common_subsequence_utf8(40345, 50022, 1)
+    result = benchmark_diff(*texts, timelimit=True)
+    speedup = f"{float(result['delta_old']) / float(result['delta_new']):.2f}"
+    print(f"text1_len: 40345; text2_len: 50022; time_new: {result['delta_new']}; time_old: {result['delta_old']}; speedup: {speedup}")
+
+    texts = gen_test_common_subsequence_utf8(12345, 15022, 5000)
+    result = benchmark_diff(*texts, timelimit=True)
+    speedup = f"{float(result['delta_old']) / float(result['delta_new']):.2f}"
+    print(f"text1_len: 12345; text2_len: 15022; time_new: {result['delta_new']}; time_old: {result['delta_old']}; speedup: {speedup}")
+
+    texts = gen_test_common_subsequence_utf8(60345, 70022, 30000)
+    result = benchmark_diff(*texts, timelimit=True)
+    speedup = f"{float(result['delta_old']) / float(result['delta_new']):.2f}"
+    print(f"text1_len: 60345; text2_len: 70022; time_new: {result['delta_new']}; time_old: {result['delta_old']}; speedup: {speedup}")
+
+    texts = gen_test_common_subsequence_utf8(60345, 70022, 55000)
+    result = benchmark_diff(*texts, timelimit=True)
+    speedup = f"{float(result['delta_old']) / float(result['delta_new']):.2f}"
+    print(f"text1_len: 60345; text2_len: 70022; time_new: {result['delta_new']}; time_old: {result['delta_old']}; speedup: {speedup}")
+
+def do_repeat():
+    results = []
+    results += benchmark_repeat(1000, 1000, timelimit=False, repeat=5000)
+    return results
+
+def do_main():
+    results = []
+    try:
+        results += benchmark_batch(10, 10, timelimit=False)
+        results += benchmark_batch(10**6, 10**6)
+        results += benchmark_batch(10**2, 10**2, timelimit=False)
+        results += benchmark_batch(10**3, 10**3, timelimit=False)
+        results += benchmark_batch(10**5, 10**5)
+        results += benchmark_batch(10**3, 10**5)
+        results += benchmark_batch(10**4, 10**5)
+        results += benchmark_batch(10**4, 10**6)
+        results += benchmark_batch(12345, 54321)
+    except Exception:
+        print("Benchmark hasn't finished successfully.")
```

Comment:
Nit: I had to change this to 
![image](https://github.com/pentesttoolscom/fast_diff_match_patch/assets/122876880/dad2759f-74d5-4fb9-9134-758f612d7106)
too see why it was not running the first time I tried it:
![image](https://github.com/pentesttoolscom/fast_diff_match_patch/assets/122876880/b23cb898-e80f-4737-a728-4ed6d07c31c8)

It seems that the solution to make it run is to uninstall the old "fast_diff_match_patch".